File: Number of paths - GFG/number-of-paths.java
//{ Driver Code Starts
import java.util.*;
import java.lang.*;
import java.io.*;

class GfG { 
    
	public static void main (String[] args) throws IOException  {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int t = Integer.parseInt(br.readLine().trim());
		while(t-->0){
		    String inputLine[] = br.readLine().trim().split(" ");
		    int M = Integer.parseInt(inputLine[0]);
		    int N = Integer.parseInt(inputLine[1]);
		    Solution ob = new Solution();
		    System.out.println(ob.numberOfPaths(M, N));
		}
	}
}




// } Driver Code Ends


class Solution{
    
    long numberOfPaths(int m, int n) {
        long ans = 1;
        long mod = 1000000007L; // Use 'L' to indicate a long literal
        for (long i = n; i <= (n + m - 2); i++) {
            ans = (ans * i) % mod;
            long invert = modInv(i - n + 1, mod);
            ans = (ans * invert) % mod;
        }
        return ans;
    }
    
    long pow(long x, int y, long mod) {
        long res = 1;
        while (y > 0) {
            if (y % 2 != 0)
                res = (res * x) % mod;

            y = y / 2;
            x = (x * x) % mod;
        }
        return res;
    }

   long modInv(long n, long mod) { 
        return pow(n, (int) (mod - 2), mod); 
    }

    
}

File: Number of paths - GFG/number-of-paths.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;
 

// } Driver Code Ends
class Solution
{
    public:
      long long pow(long long x, int y, long long mod) {
        long long res = 1;
        while (y > 0) {
            if (y % 2 != 0)
                res = (res * x) % mod;

            y = y / 2;
            x = (x * x) % mod;
        }
        return res;
    }

    long long modInv(long long n, long long mod) {
        return pow(n, static_cast<int>(mod - 2), mod);
    }
    
    
    long long  numberOfPaths(int m, int n)
    {
        long long ans = 1;
        long long mod = 1000000007LL; // Use 'LL' to indicate a long long literal
        for (long long i = n; i <= (n + m - 2); i++) {
            ans = (ans * i) % mod;
            long long invert = modInv(i - n + 1, mod);
            ans = (ans * invert) % mod;
        }
        return ans;
    }
};


//{ Driver Code Starts.

 
int main()
{
	int t;
	cin>>t;
	while(t--)
	{
		int N, M;
		cin>>M>>N;
		Solution ob;
	    cout << ob.numberOfPaths(M, N)<<endl;
	}
    return 0;
}
// } Driver Code Ends